numpy/core/fromnumeric.py

```diff
@@ -2103,7 +2110,14 @@ def all(a, axis=None, out=None, keepdims=np._NoValue):
     (28293632, 28293632, array([ True]))
 
     """
-    return _wrapreduction(a, np.logical_and, 'all', axis, None, out, keepdims=keepdims)
+    result = _wrapreduction(a, np.logical_and, 'all', axis, None, out,
+                            keepdims=keepdims)
+    if result is None:
+        return None
+    if isinstance(result, int):
```

Comment:
Sorry, but can you explain where the `int` comes from? Something like `axis is None` would make sense to me (although it gets annoying, maybe I am daft right now, but not sure there is a way to detect the reduction returns a scalar).

Reply:
I really wish we would find a way to make reductions/ufuncs scalar consistent (`axis=None` -> scalar, all others returning arrays. Scalar inputs stay scalar) :(.

Reply:
yes the _wrappereduction of ufunc returns either None, int or an array. To pass all tests need to cover all cases. Not sure if checking axis=None works the same, I would need to double check